components/script/bluetooth_blacklist.rs

```diff
@@ -77,15 +75,12 @@ fn parse_blacklist() -> Option<HashMap<String, Blacklist>> {
     // Step 1 missing, currently we parse ./resources/gatt_blacklist.txt.
     let valid_uuid_regex = Regex::new(VALID_UUID_REGEX).unwrap();
     let mut data: Vec<u8> = Vec::new();
-    let lines = read_resource_file_lines("gatt_blacklist.txt", &mut data).expect("Could not find gatt_blacklist.txt file");
+    let mut lines = read_resource_file_lines(BLACKLIST_FILE, &mut data)
+                    .expect(BLACKLIST_FILE_NOT_FOUND);
     // Step 3
     let mut result = HashMap::new();
     // Step 2 and 4
-    for line in lines {
-        let line = match line {
-            Ok(l) => l,
-            Err(_) => return None,
-        };
+    while let Some(Ok(line)) = lines.next() {
```

Comment:
We can't, because the `next()` iterator return with an `Option`. In this case with an 
`Option<Result>`.